### ISSUE TITLE
.html.erb registered as wrong filetypes

### ISSUE THREAD

**Issue by natematykiewicz** [AstroNvim/AstroNvim#1051]
### Checklist

- [X] Using a stable version of Neovim (i.e. not neovim nightly)
- [X] `:PackerSync`
- [X] `:AstroUpdate`
- [X] Restarted AstroNvim

### AstroNvim version

2.0.2

### Neovim version (>= 0.7)

0.8.0

### Operating system/version

macOS 12.5

### Describe the bug

When I open certain `.html.erb` files I notice the syntax highlighting is wrong. This is a new problem since upgrading to AstroNvim 2.0.0 with Neovim 0.8.

For a few of them if I run `:set ft`, it says `filetype=mason`. If I then `:e` or enter insert mode, the colors change, my whole bottom bar changes, and now `:set ft` returns `filetype=eruby`.

For a different named `app/views/layouts/mailer.html.erb` whose content is the following, when opening it the filetype is set to `html`. Again, running `:e` sets the filetype to `eruby`, which is the correct type.

```
<!DOCTYPE html>
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
    <style>
      /* Email styles need to be inline */
    </style>
  </head>

  <body>
    <%= yield %>
  </body>
</html>
```

In trying to debug this, I've found that if I delete the doctype line, the filetype is correctly set to `eruby`.

I do notice that the ones identified as `mason` are ones that include an ERB tag in the beginning. So, `<%= %>`.

### Steps to Reproduce

Create a file called `show.html.erb`.
Write to it `<%= %>`.
Reopen it. See the filetype is `mason`.

Now prepend `<!DOCTYPE html>` to it. Close and reopen it. See that the filetype is `html`.

### Expected behavior

The filetype should be `eruby` because the file extension is `.html.erb`.

**Comment by natematykiewicz** [comment]
I've removed all of my plugins from `~/.config/nvim/lua/user/init.lua` and still experience this. I'm unsure if it's a Neovim 0.8 problem or an AstroNvim 2.0 problem.

**Comment by natematykiewicz** [comment]
I removed all of the plugins from `~/.config/nvim/lua/user/init.lua` and still experience it. I then deleted my `~/.config/nvim/init.lua` and both examples open the file as `eruby` instead of `html` and `mason`. So something AstroNvim is doing is making it pick the wrong file type.

**Comment by mehalter** [comment]
I believe this is a bug with `mason` and how they are adding their filetype with their plugin. Luckily AstroNvim is not doing anything with how filetypes are detected so we just need to investigate what could be causing this.

**Comment by mehalter** [comment]
wow @natematykiewicz I just spend a long while debugging this and found the source in AstroNvim's code base. If I lazy load a plugin on `BufRead` or `BufReadPost` it completely borks the filetype detection in Neovim v0.8! :O

**Comment by natematykiewicz** [comment]
Crazy! So this seems like there's something wrong with Neovim 0.8?

I should also mention that in my AstroNvim config I even tried adding `html.erb` as a registered filetype (even though I was pretty sure it's already registered as that). It didn't help anything. What's really odd is that I don't even want it to automatically guess the file type based on the contents. I literally just want it to see `.html.erb` and do `rruby`.

```lua
    vim.filetype.add {
      extension = {
        jbuilder = "ruby",
        "html.erb" = "eruby", -- added this
      },
      filename = {
        ["Gemfile.lock"] = "ruby",
        ["Capfile"] = "ruby",
        ["Guardfile"] = "ruby",
        ["Vagrantfile"] = "ruby",
      },
```

**Comment by mehalter** [comment]
I am coming up with a patch fix that will circumvent this issue. Not sure if it's because of Packer or Neovim v0.8. I know a lot of refactoring has gone into `filetype.lua` in this release and this could possibly be an upstream bug. I will at least throw a bandaid on this since it's very very bad for now and then will try to dig deeper in my freetime

**Comment by natematykiewicz** [comment]
It works great!

**Comment by mehalter** [comment]
Awesome! glad it is working! I can't believe this is so weird :joy: great report!

**Comment by natematykiewicz** [comment]
Thanks for figuring it out so quickly! It seemed super weird, so I tried to be as detailed as possible.

**Comment by mehalter** [comment]
Fixed in v2.1.0
